### PR TITLE
KAFKA-17416: Add a checkstyle rule to suppress all generated code

### PR DESCRIPTION
**Pull request by xijiu**
1. remove `src/generated/java` from core module
2. add `<suppress files="[\\/](file:///)generated[\\/](file:///)" checks="[a-zA-Z0-9]*"/>` to the `suppressions.xml` to suppress all generated code

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: checkstyle/suppressions.xml [hunk 1]
```diff
@@ -237,6 +237,7 @@
     <suppress checks="ImportControl" files="FetchResponseData.java"/>
     <suppress checks="ImportControl" files="RecordsSerdeTest.java"/>
     <suppress checks="ImportControl" files="ClientTelemetryTest.java"/>
+    <suppress checks="[a-zA-Z0-9]*" files="[\\/]generated[\\/]"/>
```
Comment (chia7712) [inline checkstyle/suppressions.xml]: Please remove other generated-related suppression. they are redundant now

**Review by chia7712 — COMMENTED** [review]
@xijiu thanks for this patch

File: checkstyle/suppressions.xml (hunk 1 above)
Comment (xijiu) [reply checkstyle/suppressions.xml]: > Please remove other generated-related suppression. they are redundant now

Agree, I have already removed them.

File: checkstyle/suppressions.xml [hunk 2]
```diff
@@ -221,22 +221,7 @@
...
-    <suppress checks="(NPathComplexity|ClassFanOutComplexity|CyclomaticComplexity|ClassDataAbstractionCoupling|FinalLocalVariable|LocalVariableName|MemberName|ParameterName|MethodLength|JavaNCSS|AvoidStarImport)"
-              files="group-coordinator[\\/]build[\\/]generated[\\/].+.java$"/>
-    <suppress checks="(NPathComplexity|ClassFanOutComplexity|CyclomaticComplexity|ClassDataAbstractionCoupling|FinalLocalVariable|LocalVariableName|MemberName|ParameterName|MethodLength|JavaNCSS|AvoidStarImport)"
-              files="transaction-coordinator[\\/]build[\\/]generated[\\/].+.java$"/>
-    <suppress checks="(NPathComplexity|ClassFanOutComplexity|CyclomaticComplexity|ClassDataAbstractionCoupling|FinalLocalVariable|LocalVariableName|MemberName|ParameterName|MethodLength|JavaNCSS|AvoidStarImport)"
-              files="share-coordinator[\\/]build[\\/]generated[\\/].+.java$"/>
-
-    <suppress checks="ImportControl" files="FetchResponseData.java"/>
-    <suppress checks="ImportControl" files="RecordsSerdeTest.java"/>
-    <suppress checks="ImportControl" files="ClientTelemetryTest.java"/>
```
Comment (mumrah) [inline checkstyle/suppressions.xml]: I don't think you meant to delete these

**Comment by chia7712** [comment]
```
    <suppress checks="(NPathComplexity|ClassFanOutComplexity|CyclomaticComplexity|ClassDataAbstractionCoupling|LocalVariableName|MemberName|ParameterName|MethodLength|JavaNCSS|AvoidStarImport)"
            files="metadata[\\/]build[\\/]generated[\\/].+.java$"/>
```
```
    <suppress checks="(NPathComplexity|ClassFanOutComplexity|CyclomaticComplexity|ClassDataAbstractionCoupling|LocalVariableName|MemberName|ParameterName|MethodLength|JavaNCSS|AvoidStarImport)"
            files="clients[\\/]build[\\/]generated[\\/].+.java$"/>
```
```
    <suppress checks="(NPathComplexity|ClassFanOutComplexity|CyclomaticComplexity|ClassDataAbstractionCoupling|FinalLocalVariable|LocalVariableName|MemberName|ParameterName|MethodLength|JavaNCSS|AvoidStarImport)"
              files="core[\\/]build[\\/]generated[\\/].+.java$"/>
```
@xijiu please remove above rules also, thanks!

File: checkstyle/suppressions.xml (hunk 2 above)
Comment (xijiu) [reply checkstyle/suppressions.xml]: @mumrah  Hi David, thanks very much for code review, my fault, I will fix it.

**Comment by xijiu** [comment]
@chia7712 @mumrah    I have fixed above issues, PTAL

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**